customs.js

```diff
@@ -55,6 +57,13 @@
       }
     )
     .then(
+      (function (result) {
+        if (result.lockout) {
+          return this.db.lockAccount(account)
+        }
+      }).bind(this)
+    )
+    .then(
       function () {},
       function (err) {
         log.error({ op: 'customs.flag.1', email: email, err: err })
```

Comment:
For example, perhaps we can move the entire chain of `password.matches().then(`, `if(!match)`, `customs.flag()`, `db.lockAccount()` into a shared password-checking helper function somewhere.